### PR TITLE
collection_field_mapping removal and replacement - remove deprecation warnings

### PR DESCRIPTION
**Pull request by sephirothkod**

### CODE REVIEW

**Comment by sephirothkod** [comment]
@bkiahstroud if you want can you have a first look and see if anything sticks out as incorrect. I still haven't really looked at specs or lint/rubocop this is just a rough draft. Will continue on Monday.

File: app/factories/bulkrax/object_factory.rb [hunk 1]
```diff
@@ -223,22 +213,14 @@ def collection_type(attrs)
     # which is used by Hyrax::Actors::AddAsMemberOfCollectionsActor
     def create_attributes
       return transform_attributes if klass == Collection
-      ActiveSupport::Deprecation.warn(
-        'Creating Collections using the collection_field_mapping will no longer be supported as of Bulkrax version 3.0.' \
-        ' Please configure Bulkrax to use related_parents_field_mapping and related_children_field_mapping instead.'
-      )
-      transform_attributes.except(:collections, :collection, collection_field_mapping)
+      transform_attributes.except(:collections, :collection)
```
[the remark this answers is not in the document]
Comment (sephirothkod) [reply app/factories/bulkrax/object_factory.rb:216]: good point! will remove.

File: app/models/bulkrax/entry.rb [hunk 2]
```diff
@@ -71,11 +71,8 @@ def work_identifier
     end
 
     def self.collection_field
-      ActiveSupport::Deprecation.warn(
-        'Creating Collections using the collection_field_mapping will no longer be supported as of Bulkrax version 3.0.' \
-        ' Please configure Bulkrax to use related_parents_field_mapping and related_children_field_mapping instead.'
-      )
-      Bulkrax.collection_field_mapping[self.to_s]
+      # byebug
+      Bulkrax.related_parents_field_mapping[self.to_s]
```
[the remark this answers is not in the document]
Comment (sephirothkod) [reply app/models/bulkrax/entry.rb]: this has been replaced with a parent_field that uses a parser argument to call related_parents_paresed_mapping directly

File: app/models/bulkrax/oai_entry.rb [hunk 3]
```diff
@@ -61,6 +57,7 @@ def collections_created?
     # If OAI-PMH doesn't return setSpec in the headers for GetRecord, use parser.collection_name
     #   in this case, if 'All' is selected, records will not be added to a collection.
     def find_collection_ids
+      # TODO: replace unique_collection_identifier with related_parents_field_mapping ??
```
[the remark this answers is not in the document]
Comment (sephirothkod) [reply app/models/bulkrax/oai_entry.rb:60]: your right, I took a better look and realized that it is not really related to this work.

File: lib/bulkrax.rb [hunk 4]
```diff
@@ -35,14 +33,7 @@ class << self
...
-    #       Please configure Bulkrax to use related_parents_field_mapping and related_children_field_mapping instead.
-    # TODO: remove collection_field_mapping when releasing v2
-    # Field_mapping for establishing a collection relationship (FROM work TO collection)
-    # This value IS NOT used for OAI, so setting the OAI Entries here will have no effect
-    # The mapping is supplied per Entry, provide the full class name as a string, eg. 'Bulkrax::CsvEntry'
-    # The default value for CSV is collection
-    self.collection_field_mapping = {
+    self.related_parents_field_mapping = {
       'Bulkrax::CsvEntry' => 'collection'
     }
```
[the remark this answers is not in the document]
Comment (sephirothkod) [reply lib/bulkrax.rb]: ended up removing this entirely

File: spec/lib/bulkrax_spec.rb [hunk 5]
```diff
@@ -16,15 +16,15 @@
...
+        expect(described_class).to respond_to(:related_parents_field_mapping)
       end
-      it 'collection_field_mapping is settable' do
-        expect(described_class).to respond_to(:collection_field_mapping=)
+      it 'related_parents_field_mapping is settable' do
+        expect(described_class).to respond_to(:related_parents_field_mapping=)
       end
       it 'has a default value for CsvEntry' do
-        expect(described_class.collection_field_mapping).to eq({ 'Bulkrax::CsvEntry' => 'collection' })
+        expect(described_class.related_parents_field_mapping).to eq({ 'Bulkrax::CsvEntry' => 'collection' })
```
[the remark this answers is not in the document]
Comment (sephirothkod) [reply spec/lib/bulkrax_spec.rb]: I removed that setting completely in the end, so the specs aren't needed.

**Review by orangewolf — APPROVED** [review]

**Pull request merged**